### PR TITLE
Removed `flush_logs_every_n_steps` argument from Trainer

### PR DESCRIPTION
**Pull request by shenoynikhil**
## What does this PR do?
Part of #12521

This PR removes the `flush_logs_every_n_steps` argument in the Trainer Class and the corresponding changes. Made the following edits,

- Removed the documentation from `docs/source/common/trainer.rst` and `docs/source/visualize/logging_advanced.rst`
- Removed the `should_flush` condition in `pytorch_lightning/loops/epoch/training_epoch_loop.py`
- Removed the `flush_logs_every_n_steps` attribute from `trainer` in `pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py`

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py [hunk 1]
```diff
@@ -43,19 +43,10 @@ def __init__(self, trainer: "pl.Trainer") -> None:
...
     ) -> None:
         self.configure_logger(logger)
-        if flush_logs_every_n_steps is not None:
-            rank_zero_deprecation(
-                f"Setting `Trainer(flush_logs_every_n_steps={flush_logs_every_n_steps})` is deprecated in v1.5 "
-                "and will be removed in v1.7. Please configure flushing in the logger instead."
-            )
-        else:
-            flush_logs_every_n_steps = 100  # original default parameter
-        self.trainer.flush_logs_every_n_steps = flush_logs_every_n_steps
```
Comment (shenoynikhil) [inline pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:58]: @awaelchli I have currently removed trainer.flush_logs_every_n_steps attribute. Do you think this is suitable?

File: pytorch_lightning/loops/epoch/training_epoch_loop.py [hunk 2]
```diff
@@ -529,8 +529,7 @@ def _should_check_val_fx(self, batch_idx: int, is_last_batch: bool) -> bool:
     def _save_loggers_on_train_batch_end(self) -> None:
         """Flushes loggers to disk."""
         # this assumes that `batches_that_stepped` was increased before
-        should_flush = self._batches_that_stepped % self.trainer.flush_logs_every_n_steps == 0
```
Comment (shenoynikhil) [inline pytorch_lightning/loops/epoch/training_epoch_loop.py:532]: @awaelchli Since I have removed the `trainer.flush_logs_every_n_steps` attribute, I have made this change. Do you think this is suitable?

File: docs/source/visualize/logging_advanced.rst [hunk 3]
```diff
@@ -46,26 +46,6 @@ To change this behaviour, set the *log_every_n_steps* :class:`~pytorch_lightning
 
 ----
 
-Modify flushing frequency
```
Comment (awaelchli) [inline docs/source/visualize/logging_advanced.rst:49]: I wonder, should we keep this and change the code to TensorBoard instead (with its flushing argument)?

File: pytorch_lightning/loops/epoch/training_epoch_loop.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/loops/epoch/training_epoch_loop.py:532]: Yes, this is probably the best. We let the logger decide when to flush, so the training loop should just call the logger method.

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:58]: Yes!

**Review by awaelchli — APPROVED** [review]

File: docs/source/visualize/logging_advanced.rst (hunk 3 above)
Comment (shenoynikhil) [reply docs/source/visualize/logging_advanced.rst:49]: Should the change be part of this PR? If you think I can include it maybe.

Comment (awaelchli) [reply docs/source/visualize/logging_advanced.rst:49]: Yes if it should if we decide to do it. What do others think? @rohitgr7 @carmocca @daniellepintz

Comment (rohitgr7) [reply docs/source/visualize/logging_advanced.rst:49]: yeah! I think that'd be useful

Comment (shenoynikhil) [reply docs/source/visualize/logging_advanced.rst:49]: @rohitgr7 @awaelchli A bit confused here, should we change the code to include this argument in TensorBoard, or is the implementation present and we use the TensorBoard example here?

Comment (awaelchli) [reply docs/source/visualize/logging_advanced.rst:49]: Can't push to your branch. Here is what I wrote:


```
Modify flushing frequency
=========================

Some loggers keep logged metrics in memory for N steps and only periodically flush them to disk to improve training efficiency.
Every logger handles this a bit differently. For example, here is how to fine-tune flushing for the TensorBoard logger:

.. code-block:: python

    # Default used by TensorBoard: Write to disk after 10 logging events or every two minutes
    logger = TensorBoardLogger(..., max_queue=10, flush_secs=120)

    # Faster training, more memory used
    logger = TensorBoardLogger(..., max_queue=100)

    # Slower training, less memory used
    logger = TensorBoardLogger(..., max_queue=1)


----
```

Would you mind adding that please?

Comment (shenoynikhil) [reply docs/source/visualize/logging_advanced.rst:49]: Updated!

Comment (awaelchli) [reply docs/source/visualize/logging_advanced.rst:49]: good stuff

**Review by awaelchli — APPROVED** [review]
nice, just a minor merge conflict to address!

**Comment by shenoynikhil** [comment]
> nice, just a minor merge conflict to address!

@awaelchli Merged with master to fix the merge conflict

**Review by akihironitta — APPROVED** [review]
